apps/api-documenter/src/documenters/ExperimentYamlDocumenter.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { PackageName } from '@microsoft/node-core-library';
+import { DocComment, DocInlineTag } from '@microsoft/tsdoc';
+import { ApiModel, ApiItem, ApiItemKind, ApiDocumentedItem } from '@microsoft/api-extractor-model';
+
+import { IConfigFile, IConfigTableOfContents } from './IConfigFile';
+import { IYamlTocItem, IYamlTocFile } from '../yaml/IYamlTocFile';
+import { YamlDocumenter } from './YamlDocumenter';
+
+/**
+ * EXPERIMENTAL - This documenter is a prototype of a new config file driven mode of operation for
+ * API Documenter.  It is not ready for general usage yet.  Its design may change in the future.
+ */
+export class ExperimentYamlDocumenter extends YamlDocumenter {
+  private _config: IConfigTableOfContents;
+  private _tocPointerMap: { [key: string]: IYamlTocItem };
+  private _catchAllPointer: IYamlTocItem;
+
+  public constructor(apiModel: ApiModel, configFile: IConfigFile) {
+    super(apiModel);
+    this._config = configFile.tableOfContents!;
+
+    this._tocPointerMap = {}; // need a type?
+
+    this._generateTocPointersMap(this._config.tocConfig);
+  }
+
+  /** @override */
+  protected buildYamlTocFile(apiItems: ReadonlyArray<ApiItem>): IYamlTocFile {
+    this._buildTocItems2(apiItems);
+    return this._config.tocConfig;
+  }
+
+  private _buildTocItems2(apiItems: ReadonlyArray<ApiItem>): IYamlTocItem[] {
+    const tocItems: IYamlTocItem[] = [];
+    for (const apiItem of apiItems) {
+      let tocItem: IYamlTocItem;
+
+      if (apiItem.kind === ApiItemKind.Namespace) {
```

Comment:
>if (apiItem.kind === ApiItemKind.Namespace) { [](start = 6, length = 45)

Would this be cleaner as a `switch` satement?

Reply:
This is currently what lives in `YamlDocumenter`. A refactor of this logic is not part of this PR, although it might be something to look into in a future PR.

Reply:
That's fine.